### DISCUSSION TITLE
ZbDeviceTopic/SetOption89 unique device topic based on Zigbee device LongAddr

### DISCUSSION THREAD

**Discussion by urechi** [arendst/Tasmota#14389] [Ideas]
### ZbDeviceTopic/SetOption89

- 0 = single tele/%topic%/SENSOR topic (default)
- 1 = unique device topic based on Zigbee device ShortAddr
- 2 = unique device topic based on Zigbee device LongAddr

In my opinion, it would be nice to have this option added because if you need to replace your Zigbee coordinator or you want to change the Zigbee channel then you'll need to re-pair all your Zigbee devices. The ShortAddr will change and that requires editing your home automation configuration files with the new topics for each device. If the LongAddr would have been used instead then these extra changes won't be required on the home automation hub.
Any pro cons?

P.S Is it possible to get the LongAddr added to the mqtt topic with rules/berry or by using any other tasmota feature?

Many thanks to all tasmota contributors!

**Comment by urechi** [comment]
Also, it would be nice to make the IEEE address of the Zigbee coordinator available as a variable that can be used to compile your own version. Same like the existing %id%, %06X, %12X etc.

**Comment by sfromis** [comment]
If you really, really want the IEEE address as a topic, I'd expect it to work out if you configure the bridge with this being the topic, and use ZbName to configure the IEEE address as the "friendly" name of each device (not tested this).

My preference would be to never use the IEEE address anywhere outside of the bridge, as this makes it harder to replace stuff. Instead I'd configure topic and friendly name in an useful way, to also never use that "short address" anywhere, as it may be even worse for stability of automations.

**Reply by urechi** [reply]
Thanks for your reply @sfromis. That could be a workaround but then it would be very hard to tell what is what.

My preference would be to use the IEEE address and that's because I rename my devices more often than I swap or replace those.
